### ISSUE TITLE
Marking room as “Favourite” is not working

### ISSUE THREAD

**Issue by pReya** [element-hq/element-x-ios#3309]
### Steps to reproduce

1. Go into any rooms settings
2. Activate the “Favourite” toggle
3. Go back to room view or room list
4. Go back into the room settings
5. See: “Favourite” state is back to “off”/default state again

### Outcome

The “Favourite” status should be persisted/applied/saved

### Your phone model

iPhone 13 Mini

### Operating system version

18.0

### Application version

1.8.1

### Homeserver

Synapse 1.115

### Will you send logs?

Yes

**Comment by stefanceriu** [comment]
I can no longer reproduce this as of `1.8.5` (unreleased). Closing it for now but do feel free to reopen/comment if you still have issues.

**Comment by pReya** [comment]
> I can no longer reproduce this as of `1.8.5` (unreleased). Closing it for now but do feel free to reopen/comment if you still have issues.

Still present in 1.9.1. Can you reopen the ticket? @stefanceriu

**Comment by stefanceriu** [comment]
I still can't reproduce it, is there anything special about your stack?

**Comment by pReya** [comment]
> I still can't reproduce it, is there anything special about your stack?

No, nothing special. Synapse v1.115.0 with pretty much default settings. I can also reproduce it on a second account/phone (on the same server). So this is not specific to my account or phone. I can also see it working using a matrix.org account. So this has to be somehow server-specifc. I'll try to use some other servers, too, to see how many of them are affected. I'd be happy to share screen recording/logs/configs privately)

**Comment by stefanceriu** [comment]
It's supposed to be stored as a tag under the room account data e.g.

```
{
  "type": "m.tag",
  "content": {
    "tags": {
      "m.favourite": {}
    }
  }
}
```

There might also be some errors in the logs if you can submit a rageshake, or copy paste those lines here (settings -> report a problem -> view logs)

**Comment by stefanceriu** [comment]
Ah there we go, nice one marwing! 👏
